src/measure_theory/borel_space.lean

```diff
@@ -712,3 +713,49 @@ lemma measurable.ennnorm {f : β → α} (hf : measurable f) :
 hf.nnnorm.ennreal_coe
 
 end normed_group
+
+namespace measure_theory
+namespace measure
+
+variables [measurable_space α] [topological_space α]
+
+/-- A measure `μ` is regular if
+  - it is finite on all compact sets;
+  - it is outer regular: `μ(A) = inf { μ(U) | A ⊆ U open }` for `A` measurable;
+  - it is inner regular: `μ(U) = sup { μ(K) | K ⊆ U compact }` for `U` open. -/
+structure regular (μ : measure α) : Prop :=
+  (le_top_of_is_compact : ∀ {{K : set α}}, is_compact K → μ K < ⊤)
```

Comment:
```suggestion
  (lt_top_of_is_compact : ∀ {{K : set α}}, is_compact K → μ K < ⊤)
```
Also most other `structure`/`class` definitions don't indent the body of the definition.

A more general question: do you want it to be a `structure` or a `class`?

Reply:
I guess the mathlib way to do this is use classes. I will change it into a class.

Reply:
I'm already having second thoughts. The very first lemma that shows that a different measure is regular is
```
protected lemma smul {μ : measure α} (hμ : μ.regular) {x : ennreal} (hx : x < ⊤) :
  (x • μ).regular
```
That gets quite a bit uglier if I make it an instance. I'll keep `regular` a structure for now. I don't think it will be all that widely used anyway.